vpc-site2site-vpn/tf/variables.tf

```diff
@@ -0,0 +1,104 @@
+# ID                                     Name   Type   Length   FingerPrint          Created
+# 636f6d70-0000-0001-0000-00000014f113   pfq    rsa    4096     vaziuuZ4/BVQrgFO..   2 months ago
+variable "ssh_key_name" { }
+
+# resources created will be named: ${prefix}vpc-pubpriv, vpc name will be ${prefix} or will be defined by vpc_name
+variable "prefix" {
+  default = "vpns2s"
+}
+
+# if this is empty use the basename for the vpc name.  If not empty then use this for the vpc_name
+variable "vpc_name" {
+  default = ""
+}
+
+# zone string, us-south-1, in the example below
+# $ ibmcloud is zones
+# Listing zones in target region us-south under account Powell Quiring's Account as user pquiring@us.ibm.com...
+# Name         Region     Status   
+# us-south-3   us-south   available   
+# us-south-1   us-south   available   
+# us-south-2   us-south   available   
+
+variable "zone" {
+  default = "us-south-1"
+}
+variable "zone_bastion" {
+  # default = "us-south-1"
+  default = "eu-gb-1"
+}
+
+variable "zone_vsi" {
+  # default = "us-south-2"
+  default = "eu-gb-2"
+}
+
+# set to true if the cloud should have a public gateway.  This is used to provision software.
+variable "cloud_pgw" {
+  default = true
+}
+
+# set to true if the bastion should have a public gateway.  This is used to provision software.
+variable "bastion_pgw" {
+  default = false
+}
+# instance profile string, cc1-2x4, in the example below
+# $ ibmcloud is instance-profiles
+# Listing server profiles under account Powell Quiring's Account as user pquiring@us.ibm.com...
+# Name         Family
+# ...
+# cc1-2x4      cpu
+variable "profile" {
+  default = "cc1-2x4"
+}
+
+# image name, centos-7.x-amd64, in the example below
+# $ ibmcloud is images
+# Listing images under account Powell Quiring's Account as user pquiring@us.ibm.com...
+# ID                                     Name                    OS                                                        Created        Status   Visibility
+# cc8debe0-1b30-6e37-2e13-744bfb2a0c11   centos-7.x-amd64        CentOS (7.x - Minimal Install)                            6 months ago   READY    public
+# cfdaf1a0-5350-4350-fcbc-97173b510843   ubuntu-18.04-amd64      Ubuntu Linux (18.04 LTS Bionic Beaver Minimal Install)    6 months ago   READY    public
+# ...
+variable "image_name" {
+  default = "ubuntu-18.04-amd64"
+}
+# when true, both the frontend and backend instances will add the bastion maintenance security group
+# to their security group list, allowing ssh access from the bastion
+variable "maintenance" {
+  default = true
+}
+
+# provide the cloud-init script, empty means none
+# variable frontend_user_data { }
+# variable backend_user_data { }
+
+# the backend security group allows ingress from the frontend for this port
+# The frontend security group allows egress to the backend for this port
+variable cloud_tcp_port {
+  default = 80
+}
+
+
+##
+variable "resource_group_name" {
```

Comment:
I was not sure about the impact as that is the way it was done in the other template.  @powellquiring can you opine on this one?

Reply:
If we want to use resource_group correctly it will need to be applied to all resources individually.  At one time my understanding was the resource_group was applied to the vpc and resources in the vpc inherited.  But this is no longer true: https://ibm-cloudplatform.slack.com/archives/C8GBYJN57/p1563489456240000?thread_ts=1563485766.239200&cid=C8GBYJN57

We should document this in the solutions and recommend: `ibmcloud target -r $RG` before running tf or scripts.  Or fix up all the instances (likely not all working yet in tf).